### PR TITLE
Quantity is int-or-string

### LINKED ISSUE #94275: kubectl server side diff errors due to field type changes
**What happened**:
_Spawned from #94121_

Running kubectl diff server side errors out due to the internal data types being different.

```
# kubectl diff -f test.yaml --server-side=true
Error from server: failed to create typed patch object: errors:
  .spec.limits[0].default.cpu: expected string, got &value.valueUnstructured{Value:1}
  .spec.limits[0].defaultRequest.cpu: expected string, got &value.valueUnstructured{Value:0.5}
```

**What you expected to happen**:
No error should be shown, and the diff (if any) should be shown instead.

**How to reproduce it (as minimally and precisely as possible)**:
Given the following test manifest:

```YAML
apiVersion: v1
kind: LimitRange
metadata:
  name: default-limit-range
spec:
  limits:
  - default:
      cpu: 1
      memory: 2Gi
    defaultRequest:
      cpu: 0.5
      memory: 1Gi
    max:
      memory: 8Gi
    type: Container
```

Apply it to the cluster:
```
# kubectl apply -f test.yaml
```

Run a diff server side:
```
# kubectl diff -f test.yaml --server-side=true
Error from server: failed to create typed patch object: errors:
  .spec.limits[0].default.cpu: expected string, got &value.valueUnstructured{Value:1}
  .spec.limits[0].defaultRequest.cpu: expected string, got &value.valueUnstructured{Value:0.5}
```

**Anything else we need to know?**:
Running `kubectl apply --dry-run=server -f ...` always detects incoming changes, which might be a consequence of this issue (I can't tell for sure):

```
# kubectl apply -f test.yaml --dry-run=server
limitrange/default-limit-range configured (server dry run)
```

This issue is related to #94121, where this behavior was first observed.

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.8", GitCommit:"9f2892aab98fe339f3bd70e3c470144299398ace", GitTreeState:"clean", BuildDate:"2020-08-13T16:12:48Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.8", GitCommit:"9f2892aab98fe339f3bd70e3c470144299398ace", GitTreeState:"clean", BuildDate:"2020-08-13T16:04:18Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

- Cloud provider or hardware configuration: baremetal
- OS (e.g: `cat /etc/os-release`):
```
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

- Kernel (e.g. `uname -a`): `Linux pluto-next 5.4.0-42-generic #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by apelisse**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #94275

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

**Comment by kwiesmueller** [comment]
Is this something we should have a broader test for?
Check if this happens on other types as well?

Otherwise as you promised, not much to look at :D
/lgtm
/approve

**Comment by MonzElmasry** [comment]
@timothysc @xiang90 can you take a look on this pr ??
The code freeze is starting 12.11.2020 (about a weeks and half from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

File: staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go [hunk 1]
```diff
@@ -394,7 +394,7 @@ func (_ Quantity) OpenAPISchemaType() []string { return []string{"string"} }
 
 // OpenAPISchemaFormat is used by the kube-openapi generator when constructing
 // the OpenAPI spec of this type.
-func (_ Quantity) OpenAPISchemaFormat() string { return "" }
+func (_ Quantity) OpenAPISchemaFormat() string { return "int-or-string" }
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go:397]: we don't support this format.

And the correct schema would be `x-kubernetes-int-or-string-or-float: true`, and we don't have that.

**Pull request closed without merging**